pteranodon/plugins/abstract_plugin.py

```diff
@@ -215,6 +217,8 @@ async def _wrap_generator(gen: Callable, ret_time: float):
                     raise rpc_error
             finally:
                 await asyncio.sleep(ret_time)
+            if quit_on_error:
+                break
```

Comment:
```suggestion
            except Exception:
                if quit_on_error:
                    break
            finally:
                await asyncio.sleep(ret_time)
```

I might be forgetting how the method works but wouldn't having the `quit_on_error` check in that place make it quit once the generator has finished the first time even if there wasn't an error (but if the `quit_on_error` flag was set)? Like if you set that flag it would just exit the loop after the first time calling the generator

Reply:
You method is strictly better anyways.

The generators should stay open forever as long as there was not any error. The async for syntax keeps it open as long as it doesn't return which the MAVSDK things dont'